suite/observability/metrics_test.go

```diff
@@ -0,0 +1,66 @@
+package observability_test
+
+import (
+	"github.com/montanaflynn/stats"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/common/model"
+	"github.com/smartcontractkit/integrations-framework/environment"
+	"github.com/smartcontractkit/integrations-framework/suite/testcommon"
+	"math/rand"
+	"time"
+)
+
+var _ = Describe("OTPE metrics suite", func() {
+	var testSetup *testcommon.OCRSetupInputs
+
+	BeforeEach(func() {
+		testSetup = &testcommon.OCRSetupInputs{}
+		testcommon.NewOCRSetupInputForObservability(testSetup, 6)
+	})
+
+	It("Computes correctly the median feed price from all nodes in current round", func() {
+		By("Initializing mockserver adapter endpoints with random data")
+
+		rand.Seed(time.Now().UnixNano())
+		min := 10000
+		max := 10100
+		var adapterResults []int
+		for index := 1; index < len(testSetup.ChainlinkNodes); index++ {
+			result := rand.Intn(max-min+1) + min
+			adapterResults = append(adapterResults, result)
+		}
+		median, _ := stats.Median(stats.LoadRawData(adapterResults))
+		testcommon.SetAdapterResults(testSetup, adapterResults)
+		testcommon.SendOCRJobs(testSetup)
+
+		By("Kicking off first round")
+
+		testcommon.StartNewRound(testSetup, 1)
+
+		By("Comparing the metrics value with the expected value")
+
+		prometheus, err := environment.GetPrometheusClientFromEnv(testSetup.SuiteSetup.Env)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Eventually(func(g Gomega) int {
```

Comment:
Better to use `retry` for consistency so we don't have two libs for one purpose

Reply:
changed all retry's in tests to eventually